[SVLS-6527] Step Functions Distributed Map (#28469)
DataDog/documentation: content/en/serverless/step_functions/troubleshooting.md

## I cannot see any traces

#### Verify that your Step Function is configured to send all logs
- Ensure that the `DD_TRACE_ENABLED` tag is set to `true` on the Step Function in your AWS console.
- In your AWS console, open your Step Function's logging tab. Ensure that _Log level_ is set to `ALL`, and that _Include execution data_ is selected.
- Ensure that the CloudWatch log group (also found on the logging tab) has a subscription filter to the Datadog Lambda Forwarder in the same region.

#### Verify that logs are forwarded successfully to Datadog
- Check the Datadog Lambda Forwarder for error messages. Ensure that you have correctly set your API key and Datadog site.
- Enable `DEBUG` logs on the Datadog Lambda Forwarder by setting the environment variable `DD_LOG_LEVEL` to `debug`.

> #### Verify that logs are searchable on Live Search and have DD_TRACE_ENABLED tag

Author: Bumped these down to H4 because the sidebar feels to cluttered with all these sub-headings

<img width="1205" alt="Screenshot 2025-03-31 at 4 37 47 PM" src="https://github.com/user-attachments/assets/47c9f0a2-0126-4d77-9d70-6269ffd8bb85" />

## Some step spans are missing in the traces
- Actions from Lambda, DynamoDB, StepFunction, and most of the other AWS services are supported.
- `Wait`, `Choice`, `Success`, `Fail`, `Pass`, `Inline MapState`, and `Parallel` are supported, while [`Distributed MapState`][8] has limited support. 

> ## Search historic logs

Author: Moved this down since it's not related to troubleshooting tracing

## Some step spans are missing in the traces
- Actions from Lambda, DynamoDB, StepFunction, and most of the other AWS services are supported.
- `Wait`, `Choice`, `Success`, `Fail`, `Pass`, `Inline MapState`, and `Parallel` are supported, while [`Distributed MapState`][8] has limited support. 

## Search historic logs
To enable searching historic logs, add a temporary index to the forwarded logs. In Datadog, open the Logs [**Indexes**][3] tab. Click the **New Index** button in the upper right.

Choose a name, set the index filter to `Source:stepfunction`, leave everything else with default values, and save.

{{< img src="serverless/step_functions/log_index.png" alt="New Log index" style="width:80%;" >}}

If your organization has an existing all-encompassing index with a low limit, place your new index at the top.

**Note**: Indexing logs is not a requirement for getting traces and may incur additional cost. If you are troubleshooting a specific issue, you may wish to temporarily send logs to an index, debug, and delete the index afterwards. See [Indexes][5] for more information.

## Missing logs within an execution

> [Exclusion fiters][7] can be used to ensure all logs with the same execution_arn are either kept or excluded. Note that this has no impact on tracing.

Reviewer: ```suggestion
You can use [exclusion filters][7] to exclude a certain percentage of all logs with a particular `execution_arn`. Using exclusion filters does not impact tracing.
```

> Here is an example filter that excludes logs for 90% of the `@execution_arn`.

Reviewer: ```suggestion
In the following example, the filter excludes logs for 90% of the `@execution_arn`.
```

> {{< img src="serverless/step_functions/exclusion_filter.png" alt="New exclusion filter" style="width:80%;" >}}

Reviewer: ```suggestion
{{< img src="serverless/step_functions/exclusion_filter.png" alt="An exclusion filter named Step Functions. The 'Define exclusion query' box contains 'source:stepfunction'. Under 'Set exclusion percentage', the filter is set to exclude 90% of @execution_arn." style="width:80%;" >}}
```
